src/auth/auth.service.ts

```diff
@@ -18,208 +28,148 @@ interface OrgEmailAndId {
 export class AuthService {
   constructor(
     private readonly logger: Logger,
-    private readonly configService: ConfigService,
     private readonly tokenService: TokenService,
-    private readonly permissionService: PermissionsService,
+    private readonly permissionsService: PermissionsService,
     private readonly userSessionService: UserSessionService,
+    private readonly oidcHelperService: OidcHelperService,
+    private readonly bypassService: BypassService,
   ) {}
 
-  async getStreamlinedRegistrationToken(userId: string): Promise<string> {
-    const url = `${this.configService.get<string>(
-      'app.cdxSvcs',
-    )}/StreamlinedRegistrationService?wsdl`;
+  async determinePolicy(userId: string): Promise<PolicyResponse> {
 
-    return createClientAsync(url)
-      .then(client => {
-        return client.AuthenticateAsync({
-          userId: this.configService.get<string>('app.naasAppId'),
-          credential: this.configService.get<string>('app.nassAppPwd'),
-        });
-      })
-      .then(res => {
-        return res[0].securityToken;
-      })
-      .catch(err => {
-        if (err.root && err.root.Envelope) {
-          throw new EaseyException(
-            new Error(JSON.stringify(err.root.Envelope)),
-            HttpStatus.BAD_REQUEST,
-            { userId: userId },
-          );
-        }
-
-        throw new EaseyException(new Error(err), HttpStatus.BAD_REQUEST, {
-          userId: userId,
+    try {
+
+      /*let user: UserDTO;
```

Comment:
I'm assuming that this will get uncommented/updated with the addition of new CDX bypass logic (#6207).

Reply:
Yes, that's correct. I am going to review the bypass functionality and encapsulate as much bypass functionality as possible into a BypassService class